GoogleForCreators/web-stories-wp
# Quick Actions: UI Storybook

## Context
Part of work for quick actions in editor

## Summary

We have most of the menu we need for this done already because of what the dashboard needs, adding an icon variant with tooltip to prep for menus on editor.

## Relevant Technical Choices

Had to make a few updates to the context menu in the design system, since the functionality is the same it makes more sense to just update the styles in the few places necessary. 
- Adjust how `separator`s are added to menus (no longer a border, instead it's just `:before` of `:after`. 
- Add checks for `isIconMenu` to override default padding and spacing since icon menus need items to be squares and they have tooltips 
- Adds new prop called `isAlwaysVisible` that if true in `ContextMenu` will get passed to the menu in place of `isOpen` to allow a menu to just always be open. It also means we don't have to repurpose `isOpen` and add an additional check for the icon menus to prevent the menu's mask from rendering (since the quick action menu is always open). 

## To-do



## User-facing changes

None. Work exists in storybook. 

## Testing Instructions

Spin up storybook locally and go to the design system/components/context menu/quick action menu and see menus. 
Should match the menus listed in [figma sprint 1.7 quick actions](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7347%3A60536). Tooltips should show on focus or hover.

### QA

(From storybook, just demos of the UI - also please pardon my inconsistent screenshot size)
| Menu | Figma  | Storybook |
| -- | -- | -- |
| Nothing selected | <img width="109" alt="Screen Shot 2021-05-11 at 10 01 04 AM" src="https://user-images.githubusercontent.com/10720454/117856038-fe0ed200-b23f-11eb-96be-3d0e5235e4a9.png"> | ![Screen Shot 2021-05-10 at 5 06 17 PM](https://user-images.githubusercontent.com/10720454/117739000-2f40c100-b1b2-11eb-82cc-cd7c6fac7aa6.png) | 
| Text selected | <img width="98" alt="Screen Shot 2021-05-11 at 10 07 57 AM" src="https://user-images.githubusercontent.com/10720454/117856695-c6ecf080-b240-11eb-86d1-e3e28bf99548.png"> | ![Screen Shot 2021-05-10 at 5 06 37 PM](https://user-images.githubusercontent.com/10720454/117739003-2fd95780-b1b2-11eb-87ed-835b2d4e2840.png) |
| Shape selected | <img width="97" alt="Screen Shot 2021-05-11 at 10 07 25 AM" src="https://user-images.githubusercontent.com/10720454/117856630-b472b700-b240-11eb-89e8-8184228e3c1d.png"> | ![Screen Shot 2021-05-10 at 5 06 31 PM](https://user-images.githubusercontent.com/10720454/117739005-2fd95780-b1b2-11eb-88e4-555f585f1d54.png) | 
|  Video selected | <img width="151" alt="Screen Shot 2021-05-11 at 10 06 48 AM" src="https://user-images.githubusercontent.com/10720454/117856562-9e64f680-b240-11eb-8855-c7b0f872c9b2.png"> | ![Screen Shot 2021-05-10 at 5 06 27 PM](https://user-images.githubusercontent.com/10720454/117739007-2fd95780-b1b2-11eb-8b2c-0658d5d191f4.png) |
| Foreground image selected | <img width="128" alt="Screen Shot 2021-05-11 at 10 06 12 AM" src="https://user-images.githubusercontent.com/10720454/117856512-8e4d1700-b240-11eb-8d8f-c2d43909b435.png"> | ![Screen Shot 2021-05-10 at 5 06 24 PM](https://user-images.githubusercontent.com/10720454/117739008-2fd95780-b1b2-11eb-8508-a234c4562ae3.png) |
| Background image selected | <img width="203" alt="Screen Shot 2021-05-11 at 10 02 50 AM" src="https://user-images.githubusercontent.com/10720454/117856117-1383fc00-b240-11eb-9333-6affd34eeec3.png"> | ![Screen Shot 2021-05-10 at 5 06 20 PM](https://user-images.githubusercontent.com/10720454/117739011-3071ee00-b1b2-11eb-8eb5-ccfafeed1b2d.png)| 







This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6141
